src/python/pants/backend/go/subsystems/golang.py

```diff
@@ -84,6 +84,19 @@ class GoRoot:
     """Path to the Go installation (the `GOROOT`)."""
 
     path: str
+    version: str
+
+    def is_compatible_version(self, version: str) -> bool:
+        """Can this Go compiler handle the target version?
+
+        Inspired by
+        https://github.com/golang/go/blob/30501bbef9fcfc9d53e611aaec4d20bb3cdb8ada/src/cmd/go/internal/work/exec.go#L429-L445.
+
+        Input expected in the form `1.17`.
+        """
+        if version == "1.0":
+            return True
+        return int(version[2:]) <= int(self.version[2:])  # e.g. `1.17` -> `17`
```

Comment:
I suggest making this more robust to actually check major and minor versions. Seems like a code smell to ignore the major version here on the hope there will never be a Go v2.x.